File: spider.js
const https = require('https');
const packtUrl = 'https://www.packtpub.com/packt/offers/free-learning';

https.get( packtUrl, (res) => {
    if (res.statusCode === 200) {
        let htmlContent = '';
        res.on('data', (data) => {
            htmlContent += data;
        });
        res.on('end', ()=> {
            // The free book title is the first <h2> in the page!
            const freeBookTitle = htmlContent.split('<h2>')[1].split('</h2>')[0].trim();
            const stringToPrint = `
            The PacktPub free eBook of the day is:
            ${ '*'.repeat(freeBookTitle.length)}
            ${freeBookTitle}
            ${ '_'.repeat(freeBookTitle.length)}
            Ebook available at: ${packtUrl}
            `;
            console.log(stringToPrint);
        });
        res.on('error', (err) => {
            console.log(err);
        });
    } else {
        console.log('Something went wrong with the request!')
    }
});

File: spider.py
import requests

url = 'https://www.packtpub.com/packt/offers/free-learning'
# Without the user agent header the server responds with a 403!
user_agent = {'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/50.0.2661.102 Safari/537.36'}

response = requests.get(url, headers=user_agent)
html_content = response.text
free_book_title = html_content.split('<h2>')[1].split('</h2>')[0].strip()
print ("The PacktPub free eBook of the day is:")
print ("*" * len(free_book_title))
print (free_book_title)
print ("_" * len(free_book_title))
print ("The ebbok is available at: ", url)
